src/tasks/upgrader.ts

```diff
@@ -1,41 +1,33 @@
 const createCS = function ({ structureType, room, numberOfStructures }: any) {
 
     const structureManager = require('../managers/structures')
-    const { findStructurePosition, findValidPositionNearby } = structureManager
+    const { CSAllocationPipe } = structureManager
 
     const roomName = room.name
     const structureName = roomName + structureType
     if ((numberOfStructures === 1 || numberOfStructures === null) && (structureType == STRUCTURE_CONTAINER)) {
 
         // Merged array of SPAWNS and SOURCES
-        const structures = room.find(FIND_SOURCES).concat(room.find(FIND_MY_SPAWNS))
+        const structures = room.find(FIND_SOURCES)
 
-        // Create container near SPAWN or SOURCE
+        // Create container near SOURCE
         for (const structure in structures) {
-            let pos = findStructurePosition(structure) // pos = [x, y]
-            if (pos === null) {
-                console.log('createCSError: failed to create source', JSON.stringify({ source: structure }))
-                continue
-            } else {
-                let [x, y] = pos
-                const validPosition = findValidPositionNearby({ room, x, y })
-                if (validPosition === null || validPosition === undefined) {
-                    console.log(`createCSError: structureManager.findValidPositionNearby returned '${validPosition}'`, JSON.stringify({ source: structure }))
-                    continue
-                } else {
-                    const [_x, _y] = validPosition
-                    if (typeof _x === 'number' && typeof _y === 'number') {
-                        Game.rooms[roomName].createConstructionSite(
-                            _x,
-                            _y,
-                            structureType,
-                            structureName + numberOfStructures.toString()
-                        );
-                    }
-                }
+            const constructionSiteParams = {
+                structureToBuild: structure,
+                targetStructureToBuildAround: structureType
             }
```

Comment:
These input names are not the clearest. As I understand it in this context
- `structureToBuild` is a source and more broadly the context near what you are building
- `targetStructureToBuildAround` is the type of structure you are building.

```suggestion
            const constructionSiteParams = {
                context: structure,
                type: structureType
            }
```
```suggestion
            const constructionSiteParams = {
                adjacentElement: structure,
                type: structureType
            }
```

structureToBuild is hard to rename but it's presently unclear.

Reply:
Fair enough. Good point.

Reply:
structureToBuild is the structure we intend to build. And, targetStructureToBuildAround is the what we are building near (eg. source)